examples/long_running_measurement/_helpers.py

```diff
@@ -0,0 +1,233 @@
+"""Helper classes and functions for MeasurementLink examples."""
+
+import logging
+import pathlib
+from typing import Any, Callable, Dict, NamedTuple, TypeVar
+
+import click
+import grpc
+
+from ni_measurementlink_service import session_management
+from ni_measurementlink_service._internal.discovery_client import DiscoveryClient
+from ni_measurementlink_service._internal.stubs.ni.measurementlink.pinmap.v1 import (
+    pin_map_service_pb2,
+    pin_map_service_pb2_grpc,
+)
+from ni_measurementlink_service.measurement.service import GrpcChannelPool
+
+
+class ServiceOptions(NamedTuple):
+    """Service options specified on the command line."""
+
+    use_grpc_device: bool = False
+    grpc_device_address: str = ""
+
+    use_simulation: bool = False
+
+
+def get_service_options(**kwargs) -> ServiceOptions:
+    """Get service options from keyword arguments."""
+    return ServiceOptions(
+        use_grpc_device=kwargs.get("use_grpc_device", False),
+        grpc_device_address=kwargs.get("grpc_device_address", ""),
+        use_simulation=kwargs.get("use_simulation", False),
+    )
+
+
+T = TypeVar("T")
+
+
+def str_to_enum(mapping: Dict[str, T], value: str) -> T:
+    """Convert a string to an enum (with improved error reporting)."""
+    try:
+        return mapping[value]
+    except KeyError as e:
+        logging.error("Unsupported enum value %s", value)
+        raise grpc.RpcError(
+            grpc.StatusCode.INVALID_ARGUMENT,
+            f'Unsupported enum value "{value}"',
+        ) from e
+
+
+class PinMapClient(object):
```

Comment:
I think many of these helpers are not used by this example. Is the thought just to keep them all the same in case anyone wants to tweak this example and might make use of them later?

Reply:
`_helpers.py` is shared code, but we want each example to be a standalone program, so it is "vendored" in each example.

Maintaining N identical copies of the same file is not great, but it's better than maintaining N slightly different copies of the same file.